pkg/controller/shared_state.go

```diff
@@ -81,23 +82,35 @@ type SharedState struct {
 
 func NewSharedState(driverName, objectNamespace, topologyKey string, eventRecorder record.EventRecorder, cachedClient client.Client, azClient azdisk.Interface, kubeClient kubernetes.Interface, crdClient crdClientset.Interface) *SharedState {
 	newSharedState := &SharedState{
-		driverName:      driverName,
-		objectNamespace: objectNamespace,
-		topologyKey:     topologyKey,
-		eventRecorder:   eventRecorder,
-		cachedClient:    cachedClient,
-		crdClient:       crdClient,
-		azClient:        azClient,
-		kubeClient:      kubeClient,
-		conditionWatcher: watcher.New(context.Background(),
-			azClient, azdiskinformers.NewSharedInformerFactory(azClient, consts.DefaultInformerResync), objectNamespace),
+		driverName:             driverName,
+		objectNamespace:        objectNamespace,
+		topologyKey:            topologyKey,
+		eventRecorder:          eventRecorder,
+		cachedClient:           cachedClient,
+		crdClient:              crdClient,
+		azClient:               azClient,
+		kubeClient:             kubeClient,
+		conditionWatcher:       watcher.New(context.Background(), azClient, azdiskinformers.NewSharedInformerFactory(azClient, consts.DefaultInformerResync), objectNamespace),
 		azureDiskCSITranslator: csitranslator.NewAzureDiskCSITranslator(),
 	}
 	newSharedState.createReplicaRequestsQueue()
 
 	return newSharedState
 }
 
+func (c *SharedState) isDriverUninstall(ctx context.Context) bool {
+	w, _ := workflow.GetWorkflowFromContext(ctx)
+	crdList := &apiext.CustomResourceDefinitionList{}
+	if err := c.cachedClient.List(ctx, crdList, &client.ListOptions{}); err != nil {
+		w.Logger().Error(err, "failed to list CRDs to detect if the driver is uninstalling")
+	}
+
+	if len(crdList.Items) > 0 {
```

Comment:
It seems like we are returning `true` on `isDriverUninstall()` if we have crd in the cluster. We should only be returning true if `AzVolume` or `AzVolumeAttachment` is marked with a deletionTimestamp.